src/main/java/com/EstimatedDeliveryDate/Service/DateUtils.java

```
@@ -0,0 +1,286 @@
+		// String converted to Date
+		Date pickUpDate = null;
+		try {
+			pickUpDate = dateFormatter.parse(date);
+		} catch (ParseException e) {
+			// TODO Auto-generated catch block
+			LOGGER.error(e.getMessage());
+			e.printStackTrace();
+		}
+		Date adjustedPickUpDate = pickUpDate;
+		// Date converted to day
+		String adjustedPickUpDay = dayFormatter.format(adjustedPickUpDate).toUpperCase();
+		Integer pickUpHolidays = 0;
+		Long milliseconds = 0L;
+		// checking pick up holidays untill working day is found
+		while (checkPickUpHoliday(holidayList, adjustedPickUpDate)
+				|| checkCutOffTimeHoliday(cutOffTimeList, adjustedPickUpDay)) {
+			milliseconds = adjustedPickUpDate.getTime() + SECONDS_IN_A_DAY;
+			pickUpHolidays += 1;
+			adjustedPickUpDate = new Date(milliseconds);
+			adjustedPickUpDay = dayFormatter.format(adjustedPickUpDate).toUpperCase();
+		}
+		LOGGER.info("Number of pickup holidays" + pickUpHolidays);
+		// comparing dates to check whether pickup date is changed or not
+		if (adjustedPickUpDate.compareTo(pickUpDate) == 0) {
+			// return date in same date UTC format
+			return date;
+		}
+		// return date in UTC format
+		return utcDateFormatter.format(new Date(milliseconds));
+	}
+
+	// Adjusts the date with transit time and Processing holidays
+	public String adjustTransitHoliday(String date, Long transitTime, List<Holiday> holidayList) {
+		Long processingHolidays = 0L;
+		// String converted to UTCDate
+		Date transitDate = null;
+		try {
+			transitDate = utcDateFormatter.parse(date);
+		} catch (ParseException e) {
+			// TODO Auto-generated catch block
+			LOGGER.error(e.getMessage());
+			e.printStackTrace();
+		}
+		// String converted to Date
+		Date transitStartDate = null;
+		try {
+			transitStartDate = dateFormatter.parse(date);
+		} catch (ParseException e) {
+			// TODO Auto-generated catch block
+			LOGGER.error(e.getMessage());
+			e.printStackTrace();
```

Along with logging the error, should we throw exception to the client? Applies to other places as well.

---

But we are already throwing exception to the client when the date is not in correct format in control layer by validating ship details

---

All the exceptions should propagate to the client or swallowed with a backup logic.
I can see you are just logging the exceptions but not handling them.
The blocks of code after catch can still fail as you have not handled the nulls for dates.

---

Ok I will throw a client exception

---

Done